File: hmvvm/src/main/java/com/see99/hmvvm/model/http/interceptor/HeaderInterceptor.java
package com.see99.hmvvm.model.http.interceptor;

import android.util.Log;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * header 拦截器
 * 在header 添加参数
 */
public class HeaderInterceptor implements Interceptor {
    @Override
    public Response intercept(Chain chain) throws IOException {
        Request.Builder builder = chain.request().newBuilder();
//        Log.i("TAG", "intercept: "+chain.request().method());
//        builder.addHeader("token","111");
        return chain.proceed(builder.build());
    }
}

File: hmvvmkotlin/src/main/java/com/see99/hmvvmkotlin/model/interceptor/HeaderInterceptor.kt
package com.see99.hmvvmkotlin.model.interceptor

import okhttp3.Interceptor
import okhttp3.Response

class HeaderInterceptor : Interceptor {
    override fun intercept(chain: Interceptor.Chain): Response {
        val builder = chain.request().newBuilder()
//        Log.i("TAG", "intercept: "+chain.request().method());
//        builder.addHeader("token","111");
        //        Log.i("TAG", "intercept: "+chain.request().method());
//        builder.addHeader("token","111");
        return chain.proceed(builder.build())
    }
}